rubyforgood/human-essentials
# 4033 product drives filter by category

Resolves #4033  

### Description


Added new filter to the product drive page. This allows for filtration by item category. Made adjustments to the functions that display donation quantities and distinct item counts to return correct figures for when item category is chosen.

### Type of change



* New feature (non-breaking change which adds functionality)

### How Has This Been Tested?



Visual Test:

1. Login as organization admin.
2. Go to the product drive page.
3. Select an item category from the drop down menu.
4. Apply the filter.
5. Confirm product drive are filtered correctly and that donation quantities and distinct item counts are correct.

Rspec Tests:
Added testing to [spec/models/product_drive_spec.rb](https://github.com/rubyforgood/human-essentials/compare/main...PhantomSean:human-essentials:4033-Product-Drives-Filter-By-Category?expand=1#diff-e43ecc1991258ce0e19a3c8bb0580300bed960c8b1108a94a47eb924244ac276) to confirm that 

1. All donated organization item quantities by category and date are correctly calculated and returned.
2. All distinct donated organization items belonging to a given category are correctly counted and returned.
3. Filter `by_item_category_id` works as expected.

### Screenshots

Before:
![image](https://github.com/rubyforgood/human-essentials/assets/44501054/ba012369-5400-4217-bbeb-7a011500661d)


After:
![image](https://github.com/rubyforgood/human-essentials/assets/44501054/9ed8ffa9-9a7c-4e7f-a904-c948640dfe0a)
